### ISSUE TITLE
Question: How to support feature flags in queries.

### ISSUE THREAD

**Issue by john-twigg-ck** [apollographql/apollo-tooling#1652]
TL;DR
**Question: Does anyone have any good methods of supporting Feature Flags for portions of the queries?**

X-Post from : https://github.com/apollographql/apollo-ios/issues/900

Problem:
We're using the apollo-codegen for both iOS and Android.
There is a tight coupling between the client and the server with respect to the schema.

Example:
1) Schema is updated to v1.1
2) Server is updated to v1.1
3) Client is updated to v1.1 and starts querying for the new fields
4) Server rolls back to v1.0
**<Queries start failing>**

One option is to have a highly dynamic schema registry system on the server so as to not allow rollbacks. Let's table that idea for now.

**Question: Does anyone have any good methods of supporting Feature Flags for portions of the queries?**

Note: I know you can pass in parameters to switch off parts of the query *but they don't work with fragments as there's no parameterized fragments*   ex: `@skip(if: Boolean)` doesn't work in fragments

Ideal Example might be
```
query {
   offers { 
       ...offer
   
   }
}

fragment offer on Offer {
   apr
@if global.flagApprovalOddsEnabled
   approvalOdds 
@endif 
}
```

Challenges
- iOS and Android Code gen needs to be modified

**Comment by JakeDawkins** [comment]
Hey @john-twigg-ck! Great question, but I'm not quite sure of the best solution here (aside from not rolling back the server or not updating clients until the server is stable), but I think this discussion would be better suited for our [Spectrum Community](https://spectrum.chat/apollo). I'm sure people have certain workflows and processes for handling these things there :)
